common/blobstore/filestore/client.go

```diff
@@ -0,0 +1,129 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package filestore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/blobstore"
+	"github.com/uber/cadence/common/service/config"
+)
+
+type (
+	client struct {
+		outputDirectory string
+	}
+)
+
+// NewFilestoreClient constructs a blobstore backed by local file system
+func NewFilestoreClient(cfg *config.FileBlobstore) (blobstore.Client, error) {
+	if cfg == nil {
+		return nil, errors.New("file blobstore config is nil")
+	}
+	if len(cfg.OutputDirectory) == 0 {
+		return nil, errors.New("output directory not given for file blobstore")
+	}
+	outputDirectory := cfg.OutputDirectory
+	exists, err := common.DirectoryExists(outputDirectory)
+	if err != nil {
+		return nil, err
+	}
+	if exists {
```

Comment:
nit: I would prefer check `if !exists { // create the dir and return error}`, then `return &client{ outputDirectory: outputDirectory}, nil`